cloud/testserver/helpers.go

```diff
@@ -3,20 +3,38 @@
 
 package testserver
 
-import "io"
+import (
+	"encoding/json"
+	"io"
+	"log"
+)
 
 func write(w io.Writer, data []byte) {
-	_, _ = w.Write(data)
+	_, err := w.Write(data)
+	if err != nil {
+		log.Printf("error: %v", err)
+	}
 }
 
 func writeErr(w io.Writer, err error) {
-	_, _ = w.Write([]byte(err.Error()))
+	write(w, []byte(err.Error()))
 }
 
 func writeString(w io.Writer, str string) {
 	write(w, []byte(str))
 }
 
 func justClose(c io.Closer) {
-	_ = c.Close()
+	if err := c.Close(); err != nil {
+		log.Printf("error: %v", err)
+	}
+}
+
+func marshalWrite(w io.Writer, obj interface{}) {
+	data, err := json.Marshal(obj)
+	if err != nil {
+		log.Printf("error: %v", err)
+		return
+	}
+	write(w, data)
 }
```

Comment:
Since this function only writes JSON, I guess it's safe to assume that all callers also want to set the content-type header, we could do this here instead of at the call-site.
```suggestion
func marshalWrite(w http.ResponseWriter, obj interface{}) {
	data, err := json.Marshal(obj)
	if err != nil {
		log.Printf("error: %v", err)
		return
	}
	w.Header().Set("Content-type", "application/json")
	write(w, data)
}
```

Reply:
I follow the practice of only setting headers and the status code from the main handler function, because then it's clear from the Handler func what comes out of the API in each returning branch. Then you don't need to memorize or enter each function to know if things are set, etc.